<a href="https://colab.research.google.com/github/juanbismarck/projeto-01/blob/main/Untitled4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import kagglehub
import os
path = kagglehub.dataset_download("hubertsidorowicz/football-players-stats-2024-2025")
players=pd.read_csv(os.path.join(path, 'players_data-2024_2025.csv'))

100%|██████████| 1.17M/1.17M [00:00<00:00, 85.3MB/s]

Extracting files...


In [ ]:
somaGls=0
cont=0
fwGls=players[players['Pos']=='FW']['Gls']
for Gls in fwGls:
  somaGls+=Gls
  cont+=1
print('-'*15,'Calculando Gols','-'*15)
print(f'Média: {somaGls/cont:.2f} gols por atacante.')
mediana_gls=fwGls.median()
print(f'Mediana: {mediana_gls} gols.')
minimo_gls=fwGls.min()
print(f'Mínimo: {minimo_gls} gols.')
maximo_gls=fwGls.max()
print(f'Máximo: {maximo_gls} gols.')
desvioP_gls=fwGls.std()
print(f'Desvio Padrão: {desvioP_gls:.2f} gols.')
somaAst=0
cont=0
fwAst=players[players['Pos']=='FW']['Ast']
for Ast in fwAst:
  somaAst+=Ast
  cont+=1
print('-'*15,'Calculando assistências','-'*15)
print(f'Média : {somaAst/cont:.2f} assistências por atacante.')
mediana_ast=fwAst.median()
print(f'Mediana: {mediana_ast} assistências.')
minimo_ast=fwAst.min()
print(f'Mínimo: {minimo_ast} assistências.')
maximo_ast=fwAst.max()
print(f'Máximo: {maximo_ast} assistências.')
desvioP_ast=fwAst.std()
print(f'Desvio Padrão: {desvioP_ast:.2f} assistências.')
somaxG=0
cont=0
fwxG=players[players['Pos']=='FW']['xG']
for xG in fwxG:
  somaxG+=xG
  cont+=1
print('-'*15, 'Calculando xG','-'*15)
print(f'Média : {somaxG/cont:.2f} xG por atacante.')
mediana_xG=fwxG.median()
print(f'Mediana: {mediana_xG} xG.')
minimo_xG=fwxG.min()
print(f'Mínimo: {minimo_xG} xG.')
maximo_xG=fwxG.max()
print(f'Máximo: {maximo_xG} xG.')
desvioP_xG=fwxG.std()
print(f'Desvio Padrão: {desvioP_xG:.2f} xG.')


--------------- Calculando Gols ---------------
Média: 4.39 gols por atacante.
Mediana: 2.0 gols.
Mínimo: 0 gols.
Máximo: 28 gols.
Desvio Padrão: 5.59 gols.
--------------- Calculando assistências ---------------
Média : 1.58 assistências por atacante.
Mediana: 1.0 assistências.
Mínimo: 0 assistências.
Máximo: 18 assistências.
Desvio Padrão: 2.35 assistências.
--------------- Calculando xG ---------------
Média : 4.36 xG por atacante.
Mediana: 2.7 xG.
Mínimo: 0.0 xG.
Máximo: 24.4 xG.
Desvio Padrão: 4.95 xG.




   Com esses dados, podemos dizer que o perfil típico de um jogador ofensivo é:
* **Em relação aos gols (Gls)**: Um atacante típico marca 4.39 gols por temporada, variando 5.59 para mais ou para menos. O número de gols varia entre 0 e 28 e a mediana de 2 gols sugere que metade dos atacantes marcaram 2 gols ou menos enquanto a outra metade marcou entre 2 e 28 gols. Podemos concluir que há uma grande dispersão entre o número de gols dos atacantes, os dados são bastante heterogêneos.
* **Em relação ás assistências (Ast)**: Um atacante típico dá 1.58 assistências na temporada, variando 2.35 para mais ou para menos.A mediana sugere que metade dos jogadores deu 1 assistência ou menos enquanto a outra metade deu entre 1 e 18 assistências, o que também mostra grande dispersão no número de assistências, enquanto alguns não dão nenhuma assistência, enquanto outros podem alcançar números muito altos(18).
* **Em relaçao aos gols esperados (xG)**: Um atacante típico tem 4.36 de xG, mas a mediana sugere que metade deles tem números menores ou igual a 2.7. Também é possível identificar grande dispersão nesse parâmetro, enquanto alguns não tem nenhum gol esperado, outros têm números absurdos próximos de 24.4 gols esperados



In [3]:
print('-'*15,'Calculando média e desvio padrão de minutos por idade','-'*15)
print('-'*10,'até 20 anos','-'*10)
somaMin=0
cont=0
min020=players[players['Age']<=20]['Min']
for Min in min020:
  somaMin+=Min
  cont+=1
print(f'Média: {somaMin/cont:.2f} minutos para jogadores até 20 anos.')
desvioP_min=min020.std()
print(f'Desvio Padrão: {desvioP_min:.2f} minutos para jogadores até 20 anos.')
print('-'*10,'entre 21 e 25 anos','-'*10)
somamin=0
cont=0
min21=players['Age'] >= 21
min25=players['Age'] <= 25
min2125=players[min21 & min25]['Min']
for Min in min2125:
  somaMin+=Min
  cont+=1
print(f'Média: {somaMin/cont:.2f} minutos para jogadores entre 21 e 25 anos.')
desvioP_min=min2125.std()
print(f'Desvio Padrão: {desvioP_min:.2f} minutos para jogadores entre 21 e 25 anos.')
print('-'*10,'entre 26 e 30 anos','-'*10)
somamin=0
cont=0
min26=players['Age']>=26
min30=players['Age']<=30
min2630=players[min26 & min30]['Min']
for Min in min2125:
  somaMin+=Min
  cont+=1
print(f'Média: {somaMin/cont:.2f} minutos para jogadores entre 26 e 30 anos.')
desvioP_min=min2630.std()
print(f'Desvio Padrão: {desvioP_min:.2f} minutos para jogadores entre 26 e 30 anos.')
print('-'*10,'mais de 30 anos','-'*10)
somaMin=0
cont=0
min30=players[players['Age']>30]['Min']
for Min in min30:
  somaMin+=Min
  cont+=1
print(f'Média: {somaMin/cont:.2f} minutos para jogadores com mais de 30 anos.')
desvioP_min=min30.std()
print(f'Desvio Padrão: {desvioP_min:.2f} minutos para jogadores com mais de 30 anos.')

--------------- Calculando média e desvio padrão de minutos por idade ---------------
---------- até 20 anos ----------
Média: 507.63 minutos para jogadores até 20 anos.
Desvio Padrão: 683.84 minutos para jogadores até 20 anos.
---------- entre 21 e 25 anos ----------
Média: 1358.12 minutos para jogadores entre 21 e 25 anos.
Desvio Padrão: 929.57 minutos para jogadores entre 21 e 25 anos.
---------- entre 26 e 30 anos ----------
Média: 2543.81 minutos para jogadores entre 26 e 30 anos.
Desvio Padrão: 937.46 minutos para jogadores entre 26 e 30 anos.
---------- mais de 30 anos ----------
Média: 1391.94 minutos para jogadores com mais de 30 anos.
Desvio Padrão: 945.83 minutos para jogadores com mais de 30 anos.


Com esses dados podemos definir que:
* Para jogadores até 20 anos temos uma média baixa com alto desvio padrão, portanto os jogadores tendem a ter poucos minutos com muita variabilidade, ou seja, temos algumas joias fora da curva que recebem minutos consideráveis mas outros que ainda não estão prontos.
* Para jogadores entre 21 e 25 anos temos um aumento considerável em relação aos jogadores da faixa etária anterior, o que indica que jogadores estão mais prontos com essa idade, porém, o desvio padrão continua alto, indicando que ainda existe disparidade entre jogadores.
* Entre 26 e 30 anos temos a maior média de minutos embora o desvio seja muito próximo da faixa anterior. Esses jogadores tendem a ser os mais utilizados indicando que a maioria está no auge durante essa faixa etária.
* Para jogadores maiores de 30 anos temos uma média de minutos menor e um desvio padrão maior, provavelmente isso se deve por ser a faixa com mais jogadores e, quanto mais velho é o jogador ele tende a receber menos minutos por questões físicas.